### PR TITLE
Run "ember-cli-eslint" blueprint on "ember init"

### LINKED ISSUE #6643: ember-cli-eslint doesn't appear in the default blueprint.
Given more time to think about it, I'm not 100% happy with how `ember-cli-eslint` isn't in the default blueprint anymore. Many users don't use `ember init` to handle upgrades which means that they end up "stuck" on whichever version they have installed, or worse, delete it without paying attention as to why [when they view the comparison](https://github.com/ember-cli/ember-cli/blob/master/dev/add-to-output-repos.sh#L35).

This whole setup exists for one reason: we need to be able to run all default blueprints for bundled addons once per application creation. We may possibly in the future want to run the default blueprint for every addon in the application each time that addon gets updated. The attachment to `init` is because we can't run any blueprints unless we actually have a `node_modules` folder...

This complicates [programmatic fixture creation](https://github.com/ember-cli/ember-cli/pull/6449) as well.

I don't have a solution yet just a set of constraints and a bit of uncertainty that we've found the global maxima.

/cc @Turbo87

### PR DESCRIPTION
**Pull request by Turbo87**
... instead of running "npm install" again

Resolves #6643 (at least partly)

### CODE REVIEW

**Comment by Turbo87** [comment]
Looks like I need to fix a few tests...

File: tests/acceptance/addon-smoke-test-slow.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ describe('Acceptance: addon-smoke-test', function() {
   before(function() {
     return createTestTargets(addonName, {
       command: 'addon',
+      createESLintConfig: true,
```
Comment (twokul) [inline tests/acceptance/addon-smoke-test-slow.js]: trailing comma?

Comment (Turbo87) [inline tests/acceptance/addon-smoke-test-slow.js]: yes

https://github.com/ember-cli/ember-cli/blob/master/.eslintrc.js#L60

File: tests/acceptance/brocfile-smoke-test-slow.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ describe('Acceptance: brocfile-smoke-test', function() {
   this.timeout(500000);
 
   before(function() {
-    return createTestTargets(appName);
+    return createTestTargets(appName, { createESLintConfig: true });
```
Comment (nathanhammond) [inline tests/acceptance/brocfile-smoke-test-slow.js]: Can we name this option differently? I suspect that we will eventually want this to be `runDefaultBlueprints` or something like that.

File: tests/helpers/acceptance.js [hunk 3]
```diff
@@ -55,6 +58,18 @@ function createTestTargets(projectName, options) {
   options.command = options.command || 'new';
 
   return applyCommand(options.command, projectName, '--skip-npm', '--skip-bower', `--directory=${outputDir}`)
+    .then(() => {
+      if (options.createESLintConfig) {
```
Comment (nathanhammond) [inline tests/helpers/acceptance.js]: Everything else is deferring to the upstream Ember command, this is `mock`ing it. I'd _much_ prefer (given this is an acceptance test) to actually run the generator.

File: tests/acceptance/brocfile-smoke-test-slow.js (hunk 2 above)
Comment (Turbo87) [inline tests/acceptance/brocfile-smoke-test-slow.js]: this option does not actually run the blueprint, it just generates a very basic `.eslintrc.js` file

File: tests/helpers/acceptance.js (hunk 3 above)
Comment (Turbo87) [inline tests/helpers/acceptance.js]: I'd prefer that too, but I'm not sure how to do it as long as we use `--skip-npm` here :-/

Comment (nathanhammond) [inline tests/helpers/acceptance.js]: We use `--skip-npm` on the target creation but we link in a unified PackageCache: https://github.com/ember-cli/ember-cli/blob/master/tests/acceptance/brocfile-smoke-test-slow.js#L34

... the `npm` assets will be there.

Comment (nathanhammond) [inline tests/helpers/acceptance.js]: Hrm. We just need to do this _after_ `linkDependencies` meanwhile you were doing it at the `createTestTargets` point. I ... stand corrected. I'd like to make it actually run the blueprints, there's no particular reason that `linkDependencies` needs to be run `beforeEach` that I can tell in a quick glance while correcting my previous comment...

**Comment by kellyselden** [comment]
I might be a little out of the loop, but what do I lose if I always run `ember new my-app -sb && cd my-app && npm link ember-cli && npm i`?

**Comment by Turbo87** [comment]
@kellyselden that would be another logical step. I think I'd still prefer this PR to be merged so that we don't have to invoke npm twice.

**Comment by kellyselden** [comment]
> that would be another logical step. I think I'd still prefer this PR to be merged so that we don't have to invoke npm twice.

👍

**Comment by Turbo87** [comment]
@stefanpenner @rwjblue friendly reminder 😉

**Comment by stefanpenner** [comment]
restarting appveyor, looks like it was unable to download the commit/pr in question for node 4. Lets see if a second attempt is successful (seems likely)

**Comment by Turbo87** [comment]
@stefanpenner I had restarted that AppVeyor build multiple time in the past week and always had issues like that. It seems that AppVeyor is just broken right now... I don't see how the code changes could cause that in any way.

**Pull request merged**